realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -29,331 +35,337 @@
-                        if (!subscriber.isUnsubscribed()) {
-                            subscriber.onNext(list);
+                    public void onChange(RealmList<DynamicRealmObject> results) {
+                        if (!emitter.isCancelled()) {
+                            emitter.onNext(list);
                         }
                     }
                 };
                 list.addChangeListener(listener);
-                subscriber.add(Subscriptions.create(new Action0() {
+
+                // Cleanup when stream is disposed
+                emitter.setDisposable(Disposables.fromRunnable(new Runnable() {
                     @Override
-                    public void call() {
+                    public void run() {
                         list.removeChangeListener(listener);
                         observableRealm.close();
                         listRefs.get().releaseReference(list);
                     }
                 }));
 
-                // Immediately calls onNext with the current value, as due to Realm's auto-update, it will be the latest
-                // value.
-                subscriber.onNext(list);
+                // Emit current value immediately
+                emitter.onNext(list);
+
             }
-        });
+        }, BACK_PRESSURE_STRATEGY);
     }
 
     @Override
-    public <E extends RealmModel> Observable<E> from(final Realm realm, final E object) {
+    public <E extends RealmModel> Flowable<E> from(final Realm realm, final E object) {
         final RealmConfiguration realmConfig = realm.getConfiguration();
-        return Observable.create(new Observable.OnSubscribe<E>() {
+        return Flowable.create(new FlowableOnSubscribe<E>() {
             @Override
-            public void call(final Subscriber<? super E> subscriber) {
+            public void subscribe(final FlowableEmitter<E> emitter) throws Exception {
                 // Gets instance to make sure that the Realm is open for as long as the
                 // Observable is subscribed to it.
                 final Realm observableRealm = Realm.getInstance(realmConfig);
                 objectRefs.get().acquireReference(object);
-
                 final RealmChangeListener<E> listener = new RealmChangeListener<E>() {
                     @Override
-                    public void onChange(E object) {
-                        if (!subscriber.isUnsubscribed()) {
-                            subscriber.onNext(object);
+                    public void onChange(E obj) {
+                        if (!emitter.isCancelled()) {
+                            emitter.onNext(obj);
                         }
                     }
                 };
                 RealmObject.addChangeListener(object, listener);
-                subscriber.add(Subscriptions.create(new Action0() {
+
+                // Cleanup when stream is disposed
+                emitter.setDisposable(Disposables.fromRunnable(new Runnable() {
```

Comment:
They won't use the same instance because each `Realm.getInstance()` call is reference counted.